src/diskarray.jl

```diff
@@ -73,49 +71,65 @@ function need_batch_index(i, cs, batchstrat)
     nb = (allow_multi || has_chunk_gap(approx_chunksize.(csnow), i)) && is_sparse_index(i; density_threshold=density_thresh)
     nb, csrem
 end
-function _resolve_indices(cs, i, output_size, temp_sizes, output_indices, temp_indices, data_indices, nb)
+
+
+struct DiskIndex{N,M,A<:Tuple,B<:Tuple,C<:Tuple}
+    output_size::NTuple{N,Int}
+    temparray_size::NTuple{M,Int}
+    output_indices::A
+    temparray_indices::B
+    data_indices::C
+end
+@inline function merge_index(a::DiskIndex,b::DiskIndex)
+    DiskIndex(
+        (a.output_size...,b.output_size...),
+        (a.temparray_size...,b.temparray_size...),
+        (a.output_indices...,b.output_indices...),
+        (a.temparray_indices...,b.temparray_indices...),
+        (a.data_indices...,b.data_indices...),
+    )
+end
+
+function _resolve_indices(cs, i, indices_pre::DiskIndex, nb)
```

Comment:
Can we put a type on `nb` for all of these? would just be easier to read. Otherwise maybe spell our `nb`